### DISCUSSION TITLE
'brew cask install wireshark' command purges Wireshark v3.4.0.

### DISCUSSION THREAD

**Discussion by lademinoye** [Homebrew/discussions#171] [Everyday usage]
`brew cask install wireshark` install and remove Wireshark within the same output.  Any idea on when this will be resolved or how to resolve it? This is happening on macOS 10.15.7(19H15). I do testing and run this command weekly, just started noticing this issue last week.

```
IG-11:~ nxs$ brew cask install wireshark
==> Tapping homebrew/cask
Cloning into '/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask'...
remote: Enumerating objects: 18, done.
remote: Counting objects: 100% (18/18), done.
remote: Compressing objects: 100% (16/16), done.
remote: Total 494754 (delta 8), reused 4 (delta 2), pack-reused 494736
Receiving objects: 100% (494754/494754), 226.99 MiB | 1.61 MiB/s, done.
Resolving deltas: 100% (351251/351251), done.
Tapped 1 command and 3728 casks (3,846 files, 243MB).
==> Downloading https://2.na.dl.wireshark.org/osx/Wireshark%203.4.0%20Intel%2064
######################################################################## 100.0%
==> Verifying SHA-256 checksum for Cask 'wireshark'.
==> Installing Cask wireshark
==> Running installer for wireshark; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are i
Password:
installer: Package name is Add Wireshark to the system PATH
installer: Upgrading at base path /
installer: The upgrade was successful.
==> Running installer for wireshark; your password may be necessary.
==> Package installers may write to any location; options such as --appdir are i
installer: Package name is ChmodBPF
installer: Upgrading at base path /
installer: The upgrade was successful.
==> Moving App 'Wireshark.app' to '/Applications/Wireshark.app'.
==> Backing App 'Wireshark.app' up to '/usr/local/Caskroom/wireshark/3.4.0/Wires
==> Removing App '/Applications/Wireshark.app'.
==> Purging files for version 3.4.0 of Cask wireshark
Error: It seems there is already a Binary at '/usr/local/bin/editcap'.
```

**Comment by gromgit** [comment]
> This is happening on macOS 10.15.7(19H15) as well as Ubuntu 20.04 LTS.

Casks aren't supported on Linux, so it's surprising that you managed to replicate this issue in Ubuntu. :grin: 

I've just [filed a related bug in `brew`](https://github.com/Homebrew/brew/issues/9288), but the root problem is that the `wireshark` formula and cask _actually conflict_ as of Oct 30 2020, so you should've been prevented from installing the cask. If you need to use any of the stuff in the cask, install _only_ the cask, since it seems to also deploy the command-line tools that you'd get with the formula.

**Reply by lademinoye** [reply]
Thanks for the reply. And I misspoke about the Ubuntu 20.04 LTS.

**Reply by RedClouud** [reply]
This was my problem. I just needed to uninstall the wireshark formula and install the cask to resolve the conflicts. (idk what the difference is but it worked):
`brew remove wireshark`
then
`brew install --cask wireshark`

**Reply by p-linnane** [reply]
A formula is a binary we build from upstreams distributed source code. A cask just fetches the binary upstream provides.

**Comment by jackjansen** [comment]
Just a note that this issue still exists as of January 2022.

The workaround suggested by @gromgit (`brew uninstall wireshark` before `brew install --cask wireshark`) worked for me.

**Reply by MonFireE** [reply]
still exists as of March 2023

**Reply by p-linnane** [reply]
https://github.com/Homebrew/homebrew-cask/issues/12822

**Reply by missinglink** [reply]
I ran `brew install wireshark` first, then `brew install --cask wireshark` which resulted in `Error: It seems there is already a Binary at '/opt/homebrew/bin/idl2wrs'.`.

Following the suggestion above, removing the CLI tools *before* installing the GUI tools worked for me.

```
brew --version
Homebrew 4.1.15
```

```
uname -a
Darwin mac 22.5.0 Darwin Kernel Version 22.5.0: Thu Jun  8 22:29:26 PDT 2023; root:xnu-8796.121.3~8/RELEASE_ARM64_T8112 arm64 arm Darwin
```

**Reply by NUAAMETEOR** [reply]
Thanks,it works

**Comment by amiran-malania** [comment]
Still had this issue. None of the above worked for me. Eventually did rm -rf on Applicaitons/wireshark directory

**Comment by mianmuhammadse** [comment]
The issue still persists but the workaround mentioned by @gromgit still works.

**Comment by chanthos-selvaraj** [comment]
The issue still persists but the workaround mentioned by @gromgit still works.
